src/main/java/net/minestom/server/item/ItemMetaBuilder.java

```diff
@@ -0,0 +1,155 @@
+package net.minestom.server.item;
+
+import net.kyori.adventure.text.Component;
+import net.minestom.server.item.attribute.ItemAttribute;
+import net.minestom.server.utils.NBTUtils;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jglrxavpok.hephaistos.nbt.NBTCompound;
+
+import java.util.*;
+import java.util.function.Supplier;
+
+public abstract class ItemMetaBuilder implements Cloneable {
+
+    protected int damage;
+    protected boolean unbreakable;
+    protected int hideFlag;
+    protected Component displayName;
+    protected List<Component> lore = new ArrayList<>();
+    protected Map<Enchantment, Short> enchantmentMap = new HashMap<>();
+    protected List<ItemAttribute> attributes = new ArrayList<>();
+    protected int customModelData;
+
+    protected NBTCompound originalNBT;
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder damage(int damage) {
+        this.damage = damage;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder unbreakable(boolean unbreakable) {
+        this.unbreakable = unbreakable;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder hideFlag(int hideFlag) {
+        this.hideFlag = hideFlag;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder hideFlag(@NotNull ItemHideFlag... hideFlags) {
+        int result = 0;
+        for (ItemHideFlag hideFlag : hideFlags) {
+            result |= hideFlag.getBitFieldPart();
+        }
+        return hideFlag(result);
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder displayName(@Nullable Component displayName) {
+        this.displayName = displayName;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder lore(@NotNull List<@NotNull Component> lore) {
+        this.lore = lore;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder lore(Component... lore) {
+        lore(Arrays.asList(lore));
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder enchantments(@NotNull Map<Enchantment, Short> enchantments) {
```

Comment:
`@NotNull` on `Enchantment` and the `Short`. Could we use a list of a new class, `EnchantmentLevel` that holds a primitive short?

Reply:
What is the goal? I believe that `Map#of` should be enough